src/components/landing/UserLogin.js

```diff
@@ -20,23 +24,68 @@ class UserLogin extends Component {
   async handleSubmit(event) {
     event.preventDefault();
     const { username, password } = this.state;
-    if (username === "" || password === "") {
+
+    if (username === '' && password === '') {
+      this.setState({ doShake: true });
       return;
     }
 
     const auth = new ServiceResolver().AuthResolver();
-    const jwtToken = await auth.loginMember({ username, password });
-    SessionStorageHelper.StoreJwt(jwtToken);
+
+    try {
+      this.setState({ enabled: false });
+
+      const jwtToken = await auth.loginMember({ username, password });
+      SessionStorageHelper.StoreJwt(jwtToken);
+
+      this.setState({
+        enabled: false,
+        doShake: false,
+        loginSuccess: true
+      });
+    } catch {
+      this.setState({
+        enabled: true,
+        doShake: true
+      });
+
+      setTimeout(() => {
+        this.setState({ doShake: false });
+      }, 200);
+    }
   }
 
   handleChange(event) {
     const name = event.target.name;
-    this.setState({ [name]: event.target.value });
+    this.setState(
+      {
+        [name]: event.target.value,
+        errorMessage: null
+      },
+      () => {
+        if (this.state.username === '' || this.state.password === '') {
+          this.setState({ enabled: false });
+        } else {
+          this.setState({ enabled: true });
+        }
+      }
+    );
   }
 
   render() {
+    const { enabled, doShake, loginSuccess } = this.state;
+    let loginFormClassName = 'login-form mt-4';
+
+    if (doShake) {
+      loginFormClassName += ' shake-horizontal';
+    }
+
+    if (loginSuccess) {
+      return <Redirect to="/profile" />;
```

Comment:
We should redirect to the member profile that is currently authenticated. You can define a memberId variable and grab the id by decoding the jwt token.
You can grab the id with this code. You will have to import `JwtTokenHelper` and `SessionStorageHelper`
`const memberId = JwtTokenHelper.decodeMemberJwt(SessionStorageHelper.GetJwt().token).id;`
```suggestion
      return <Redirect to=`/members/{memberId}` />;
```